core-bundle/src/Resources/contao/config/constants.php

```diff
@@ -8,16 +8,19 @@
  * @license LGPL-3.0-or-later
  */
 
+// Backwards compatibility
 // Core version
 define('VERSION', '4.13');
 define('BUILD', '0-dev');
 define('LONG_TERM_SUPPORT', false);
 
+// Backwards compatibility
 // Link constants
 define('LINK_BLUR', ' onclick="this.blur()"');
 define('LINK_NEW_WINDOW', ' onclick="return !window.open(this.href)"');
 define('LINK_NEW_WINDOW_BLUR', ' onclick="this.blur();return !window.open(this.href)"');
 
+// Backwards compatibility
```

Comment:
May be should put these constants in a class somewhere? e.g. `ContaoContext` - since at least some of those are used when pushing entries into the system log for example. Then we can deprecate them and remove them from Contao 5.0.

Reply:
I would rather replace them with the string values directly.

Reply:
But they have special meaning which could be documented in the code. e.g. if you use `TL_ERROR` (`'ERROR') a system log entry will be coloured red. `TL_CRON` (`'CRON'`) entries will be coloured green etc.

Reply:
Ah, now I understand I think. The class `Contao\CoreBundle\Monolog\ContaoContext` already exists and has all these these constants already. I will change the usage everywhere in the core then I think. Is that what you meant? ☺️

Reply:
Right, I wasn't even really aware that these constants are already present there 😁. Yes, that is what I meant 👍

Reply:
On that note: since `System::log` is also deprecated - you would proabably need to change any `$this->log` or `System::log` occurrence to:

```php
System::getContainer()->get('logger')->log('…', ['contao' => new ContaoContext(__METHOD__, ContaoContext::GENERAL)
```

However, I personally always find this rather cumbersome and still often use just `System::log`, since it's more convenient. May be we could introduce a `SystemLogger` helper service to improve DX here? Follow up PR of course, since out of scope.

Reply:
Done in c67d824ebb32d37d2df2ba7f622db59f494470af

> Follow up PR of course, since out of scope.

Created issue #3701